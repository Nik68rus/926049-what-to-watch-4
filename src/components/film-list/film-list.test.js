import React from 'react';
import renderer from 'react-test-renderer';
import FilmList from './film-list.jsx';

const filmList = [{
  id: 1,
  src: `test.src`,
  preview: `test.preview`,
  title: `Test movie`,
  poster: `test.poster`,
  background: `test.background`,
  backgroundColor: `test.color`,
  genre: `test.genre`,
  date: 0,
  rating: 0,
  rateCount: 0,
  descriptiom: `test.description`,
  director: `test.director`,
  actors: [`test.actor1`],
  runTime: 0,
  isFavorite: false,
}, {
  id: 2,
  src: `test.src`,
  preview: `test.preview`,
  title: `Test movie`,
  poster: `test.poster`,
  background: `test.background`,
  backgroundColor: `test.color`,
  genre: `test.genre`,
  date: 0,
  rating: 0,
  rateCount: 0,
  descriptiom: `test.description`,
  director: `test.director`,
  actors: [`test.actor1`],
  runTime: 0,
  isFavorite: false,
}];

it(`FilmList rendered correctly`, () => {
  const tree = renderer.create(<FilmList films={filmList} onCardClick={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
