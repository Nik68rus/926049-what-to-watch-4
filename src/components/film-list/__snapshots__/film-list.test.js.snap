// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList rendered correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="test.preview"
        src="test.src"
      />
      <img
        alt="Test movie"
        height="175"
        src="test.preview"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Test movie
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        poster="test.preview"
        src="test.src"
      />
      <img
        alt="Test movie"
        height="175"
        src="test.preview"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Test movie
      </a>
    </h3>
  </article>
</div>
`;
